snowexsql/tables/doi.py

```diff
@@ -10,7 +10,7 @@ class DOI(Base):
     """
     __tablename__ = 'dois'
 
-    doi = Column(String())
+    doi = Column(String(), nullable=False, index=True)
```

Comment:
If I understand correctly, this enforces that any class that inherits HasDOI has a non-null DOI entry. I think we agreed that is true for 'sites', but is it also true for 'campaign_observations'? Actually, why does campaign_observations need the DOI when sites already requires campaign_id which is required in campaign_observations? Just checking if this is a duplication but I might be missing something.

Reply:
Thanks for clarifying that - it makes total sense now.